shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm

```diff
@@ -128,60 +87,18 @@
 
   FML_DCHECK(platform_view) << "Platform view not found for id: " << platform_view_id;
 
-  CGFloat scale = [[NSScreen mainScreen] backingScaleFactor];
+  CGFloat scale = platform_view.layer.contentsScale;
   platform_view.frame = CGRectMake(layer->offset.x / scale, layer->offset.y / scale,
                                    layer->size.width / scale, layer->size.height / scale);
   if (platform_view.superview == nil) {
     [default_base_view addSubview:platform_view];
+    [default_base_view.layer addSublayer:platform_view.layer];
   }
   platform_view.layer.zPosition = layer_position;
 }
 
-void FlutterCompositor::SetPresentCallback(
-    const FlutterCompositor::PresentCallback& present_callback) {
-  present_callback_ = present_callback;
-}
-
-void FlutterCompositor::StartFrame() {
-  // First remove all CALayers from the superlayer.
-  for (auto layer : active_ca_layers_) {
-    [layer removeFromSuperlayer];
-  }
-
-  // Reset active layers.
-  active_ca_layers_.clear();
-  SetFrameStatus(FrameStatus::kStarted);
-}
-
-bool FlutterCompositor::EndFrame(bool has_flutter_content) {
-  bool status = present_callback_(has_flutter_content);
-  SetFrameStatus(FrameStatus::kEnded);
-  return status;
-}
-
 FlutterView* FlutterCompositor::GetView(uint64_t view_id) {
```

Comment:
We can actually remove this function (replace it with its body) now that `FlutterCompositor` doesn't have subclasses any more.